calico/openstack/t_etcd.py

```diff
@@ -276,9 +276,17 @@ def atomic_delete_endpoint(self, endpoint):
             endpoint.id,
             endpoint.modified_index
         )
-        self.client.delete(
-            endpoint.key, prevIndex=endpoint.modified_index, timeout=5
-        )
+
+        try:
+            self.client.delete(
+                endpoint.key, prevIndex=endpoint.modified_index, timeout=5
+            )
+        except etcd.EtcdKeyNotFound:
```

Comment:
'This method allows exceptions from etcd to bubble up' contradicts this (slightly).  Please adjust the comment.